### PR TITLE
update ci.yml to test Nim 2.2; also test gcc-14

### PR DESCRIPTION
**Pull request by narimiran**

### CODE REVIEW

**Comment by metagn** [comment]
The Windows failure happens in `test_io` after Nim commit https://github.com/nim-lang/Nim/commit/8f5ae28fab113c425d22e5abc230f456fa627744. The issue seems to be with clang's stack protection, the new generated code breaks it in some way that causes `__chkstk_ms` to be called repeatedly. Changing the compiler options to as little as `--passC:"-O1"` makes it work, but none of the switches I tried that disabled the stack protection worked. `-d:release` and even `--debugger:gdb` also makes it work but `--linedir:on` which `--debugger:gdb` enables doesn't.

**Comment by tersec** [comment]
> The Windows failure happens in `test_io` after Nim commit [nim-lang/Nim@8f5ae28](https://github.com/nim-lang/Nim/commit/8f5ae28fab113c425d22e5abc230f456fa627744). The issue seems to be with clang's stack protection, the new generated code breaks it in some way that causes `__chkstk_ms` to be called repeatedly. Changing the compiler options to as little as `--passC:"-O1"` makes it work, but none of the switches I tried that disabled the stack protection worked. `-d:release` and even `--debugger:gdb` also makes it work but `--linedir:on` which `--debugger:gdb` enables doesn't.

To the extent this is a clang issue, what we've done before on these is check if newer versions of clang have fixed it -- if so, can note that and also see if there's some workaround in the meantime. But especially on Windows, clang isn't a native compiler tied to the OS, so it's ok to just require newer clang versions.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -14,21 +14,31 @@ jobs:
...
           - os: macos
             cpu: amd64
+          - os: macos
+            cpu: arm64
           - os: windows
             cpu: amd64
-          #- os: windows
-            #cpu: i386
-        branch: [version-1-6, version-2-0, devel]
+        branch: [version-2-0, version-2-2, devel]
```
Comment (arnetheduck) [inline .github/workflows/ci.yml:27]: as with the other low-level libraries, this should test 1.6 as well (in particular, it should test what the nimble file specifies as compatible)

**Comment by arnetheduck** [comment]
obsolete by common workflow, should also test 1.6 etc

**Pull request closed without merging**